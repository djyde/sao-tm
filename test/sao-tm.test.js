const path = require('path')
const sao = require('sao')

describe('sao-tm', async () => {
  it('should not include React', async () => {
    const mockPromptAnswers = {
      name: 'foo',
    }
    const generator = path.join(__dirname, '..')
    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(await stream.readFile('README.md')).toMatchSnapshot()
    expect(await stream.readFile('package.json')).toMatchSnapshot()
  })

  it('should include React', async () => {
    const mockPromptAnswers = {
      name: 'foo',
      isReact: true,
      isNode: false
    }
    const generator = path.join(__dirname, '..')
    const stream = await sao.mock({ generator }, mockPromptAnswers)

    expect(await stream.readFile('README.md')).toMatchSnapshot()
    expect(await stream.readFile('package.json')).toMatchSnapshot()
  })
})