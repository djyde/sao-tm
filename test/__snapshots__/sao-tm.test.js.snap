// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sao-tm should include React 1`] = `
"# foo

[![NPMVersion](https://badgen.net/npm/v/foo)](https://npm.im/foo)

> my stunning project

## License

MIT &copy; mock_username
"
`;

exports[`sao-tm should include React 2`] = `
"{
  \\"name\\": \\"foo\\",
  \\"version\\": \\"0.0.1\\",
  \\"script\\": {
    \\"dev\\": \\"tsc --watch\\",
    \\"build\\": \\"tsc\\",
    \\"test\\": \\"npm run build && jest\\",
    \\"prepublish\\": \\"npm run test\\"
  },
  \\"dependencies\\": {
    
      \\"react\\": \\"^16\\",
      \\"react-dom\\": \\"^16\\"
    
  },
  \\"devDependencies\\": {
    
    
      \\"@types/react\\": \\"^16\\",
      \\"@types/react-dom\\": \\"^16\\",
    
    \\"typescript\\": \\"^3\\",
    \\"jest\\": \\"^23\\"
  }
}"
`;

exports[`sao-tm should not include React 1`] = `
"# foo

[![NPMVersion](https://badgen.net/npm/v/foo)](https://npm.im/foo)

> my brilliant project

## License

MIT &copy; mock_username
"
`;

exports[`sao-tm should not include React 2`] = `
"{
  \\"name\\": \\"foo\\",
  \\"version\\": \\"0.0.1\\",
  \\"script\\": {
    \\"dev\\": \\"tsc --watch\\",
    \\"build\\": \\"tsc\\",
    \\"test\\": \\"npm run build && jest\\",
    \\"prepublish\\": \\"npm run test\\"
  },
  \\"dependencies\\": {
    
  },
  \\"devDependencies\\": {
    
      \\"@types/node\\": \\"^11\\",
    
    
    \\"typescript\\": \\"^3\\",
    \\"jest\\": \\"^23\\"
  }
}"
`;
